TACC/Core-Styles
# feat!: ease use of core-styles

## Overview

Make Core-Styles easier to use (for clients and devs).

<sup>Initially, for TUP-UI.  Conveniently, for Core-CMS. Eventually, for TAM and CEPv3.<sup>

## Related

- [TUP-372](https://jira.tacc.utexas.edu/browse/TUP-372)
- builds atop https://github.com/TACC/Core-Styles/pull/78
- required by https://github.com/TACC/Core-CMS/pull/569
- required by https://github.com/TACC/tup-ui/pull/70

## Changes

- changed demo index page to:
    - have the word "Library" in the title
    - reference latest client list
    - add links to important CSS docs
- ❗️ **changed context data API:**
    - `styles.shouldSkipBase` → `shouldSkipPattern`
    - `styles.….global` → `globalStyles`
- ❗️ **changed font replacement to be predictable downstream**
- ❗️ **removed partial-based Bootstrap and CMS asset load**
- ❗️ **changed asset load order and layering to be accurate to client needs**
- added global pattern-asset-load debug panel to demo
- changed pattern configs to:
    - use new context data API
    - not load patterns already loaded by `core-styles.base.css`
- removed duplicate demo styles
- changed `core-styles.base.css` to load `core-styles.settings.css`
- added `core-styles.demo.css` (and internal stylesheet)
- fixed `<a>` and `<a class="c-button…">` conflict
- fixed `<button class="c-button…">` and `<a class="c-button…">` to have same widths
- fixed `o-table-wrap` incomplete usage of `c-message`
- fixed `s-affixed-input-wrapper` style conflict with `core-styles.demo.css`
- fixed missing style to prevent conflict between Bootstrap and CMS/Portal Nav styles

## Testing

1. Build and serve demo.
2. Verify all patterns load.
3. Verify no patterns are broken.

## UI

For feature showcase and pattern testing, see [(Confluence) Shared UI - CSS - Core Styles](https://confluence.tacc.utexas.edu/x/q5V9E) at "Has a Demo".

## Notes

Beyond allowing other teams to use Core-Styles with so much less friction, this is also a major quality-of-life improvement for Core-Styles developers—so much so that I feel Core-Styles is ready for other developers.
